src/apps/Permissions/AppCard.js

```diff
@@ -17,8 +17,13 @@ class AppCard extends React.PureComponent {
   }
   render() {
     const { app } = this.props
-    const { name, identifier, proxyAddress } = app
-    const instanceLabel = identifier || shortenAddress(proxyAddress)
+    // Note: We could also use app.isAragonOsInternalApp,
+    // but we can just merge system and contract-only apps
+    // so the user doesn't get lost in a sea of app types
+    const { name, identifier, hasWebApp, proxyAddress } = app
+    const instanceLabel = !hasWebApp
```

Comment:
I don't know if this is the model we want to pursue, especially for apps where multiple contracts may hook up to one frontend. Perhaps we could have three categories:

- System apps: "internal" apps
- Background apps: "no frontend" apps
- Normal apps

Reply:
I like this model, having System and Background apps

Reply:
Implemented it, ready to review @sohkai

Reply:
LGTM